compiler/src/dotty/tools/dotc/core/Denotations.scala

```diff
@@ -798,8 +798,6 @@ object Denotations {
           assert(!d.is(Package), s"illegal transformation of package denotation by transformer $transformer")
         case _ =>
 
-      def escapeToNext = nextDefined.ensuring(_.validFor != Nowhere)
-
```

Comment:
`ensuring` is defined using an implicit class: https://github.com/scala/scala/blob/4905801fc161300ba20bae4f7adf045b1698e831/src/library/scala/Predef.scala#L356-L361. Could we define it using extension methods in Scala 3 instead? It should ideally be a no-op when assertions are disabled.